### PR TITLE
Make batching task finite by adding CLOSE_EVENT_SENTINEL

### LINKED ISSUE #8070: Presumably finished realization displayed as pending in joblist view
A user has reported the following issue when running ~~scheduler~~ JobQueue.

--- 

Hi.. 

Currently I am experiencing several realizations finished without doing the steps in the configuration and it is labeled as 'finished'.

 

I know sometimes it is common to have some error due to maybe random licensed error, etc. therefore I set max_submit = 3 just to make sure that if something like that happened (failed run), the case will be restarted.

 

However if the label is 'finished' I think the case wont be restarted. in some cases it can be more than 5 realization like that (picture attached)

Image

### PR DESCRIPTION
**Pull request by xjules**
**Issue**
~~Resolves https://github.com/equinor/ert/issues/8359~~
Resolves #8070 

**Approach**
That error might be due to the last batch of events not being processed. Therefore adding SENTINEL object.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [x] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/ensemble_evaluator/evaluator.py [hunk 1]
```diff
@@ -271,23 +279,10 @@ async def handle_dispatch(self, websocket: WebSocketServerProtocol) -> None:
                             f"ignoring since I am {self.ensemble.id_}"
                         )
                         continue
-                    try:
-                        if event["type"] == EVTYPE_FORWARD_MODEL_CHECKSUM:
-                            await self.forward_checksum(event)
-                        else:
-                            await self._events.put(event)
-                    except BaseException as ex:
```
Comment (berland) [inline src/ert/ensemble_evaluator/evaluator.py:279]: This should be a separate commit (you choose if also a different PR), to be able to have a commit message properly explaining why.

Comment (xjules) [reply src/ert/ensemble_evaluator/evaluator.py:279]: Ok, can remove it from this PR.

File: src/ert/ensemble_evaluator/evaluator.py [hunk 2]
```diff
@@ -56,6 +56,8 @@
 
 EVENT_HANDLER = Callable[[List[CloudEvent]], Awaitable[None]]
 
+CLOSE_EVENT_SENTINEL = object()
```
Comment (berland) [inline src/ert/ensemble_evaluator/evaluator.py:59]: Should we make this into a dummy class, so that we can type the queue more properly than  `Any`

**Comment by xjules** [comment]
Closing this one as this seems not to fix the underlying issue with PENDING realizations.
